tests/Fixer/Import/NoUnusedImportsFixerTest.php

```diff
@@ -1319,42 +1319,80 @@ public function foo() {}
         ];
 
         yield [
-            $expected = <<<'EOF'
+            <<<'EOF'
+                <?php
+                use some\a\{ClassD};
+                use function some\c\{fn_a};
+                use const some\d\{ConstB};
+
+                new CLassD();
+                echo fn_a(ConstB);
+                EOF,
+            <<<'EOF'
                 <?php
                 use some\a\{ClassD};
                 use some\b\{ClassA, ClassB, ClassC as C};
                 use function some\c\{fn_a, fn_b, fn_c};
                 use const some\d\{ConstA, ConstB, ConstC};
 
                 new CLassD();
-                echo fn_a();
+                echo fn_a(ConstB);
                 EOF
         ];
 
-        yield [ // TODO test shows lot of cases where imports are not removed while could be
-            '<?php use A\{B,};
-use some\y\{ClassA, ClassB, ClassC as C,};
-use function some\a\{fn_a, fn_b, fn_c,};
-use const some\Z\{ConstAA,ConstBB,ConstCC,};
-use const some\X\{ConstA,ConstB,ConstC,ConstF};
-use C\{D,E,};
+        $empty = '';
 
-    echo ConstA.ConstB.ConstC,ConstF;
-    echo ConstBB.ConstCC;
-    fn_a(ClassA::test, new C());
-',
-            '<?php use A\{B,};
-use some\y\{ClassA, ClassB, ClassC as C,};
-use function some\a\{fn_a, fn_b, fn_c,};
-use const some\Z\{ConstAA,ConstBB,ConstCC,};
-use const some\X\{ConstA,ConstB,ConstC,ConstF};
-use C\{D,E,};
-use Z;
-
-    echo ConstA.ConstB.ConstC,ConstF;
-    echo ConstBB.ConstCC;
-    fn_a(ClassA::test, new C());
-',
+        yield 'grouped imports' => [
+            <<<EOF
+                <?php
+                use some\\y\\{ClassA, ClassC as C,};
+                use function some\\a\\{
+                    {$empty}
+                    fn_b,
+                    {$empty}
```

Comment:
This is last failing test case, I found it problematic and wanted to discuss it. First of all: should we expect fixer to clean up empty lines? If yes, I'll try to find a good way (there are edge cases, because if there's a comment the line should be kept as-is). If not, can anyone point me to good example how to handle trailing whitespace in string? There's probably also a issue with merging surrounding whitespace during clean-up. Anyway, it's last(?) thing to do here.

Reply:
Fixed in [1cc34191](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7815/commits/1cc34191d1fd3b9dc45c4441341760b081226ebd).